src/hessenberg.jl

```diff
@@ -0,0 +1,39 @@
+import Base.LinAlg: Givens, givensAlgorithm
+
+mutable struct Hessenberg{T<:AbstractMatrix}
+    H::T # H is assumed to be Hessenberg of size (m + 1) × m
+end
+
+@inline Base.size(H::Hessenberg, args...) = size(H.H, args...)
+
+function solve!(H::Hessenberg, rhs)
```

Comment:
I think we should stick with `A_ldiv_B!` here even though it invalidates the factorization. Probably a good idea to add a comment, though.